### ISSUE TITLE
Bug(sort-imports): Invalid newlines auto-fix under very special conditions    

### ISSUE THREAD

**Issue by hugop95** [azat-io/eslint-plugin-perfectionist#329]
### Describe the bug

Newline auto-fixes can in very specific configurations result in code that does not compile. See code example.

### Code example

Configuration:

`"newlinesBetween": "always"` or `"never"`.

Code:

```ts
import a from 'a'

 import b from 'b'
```

Notice the whitespace before the second import.

The resulting code is:
```ts
import a from 'a' import b from 'b'
```

### ESLint version

8.57.0

### ESLint Plugin Perfectionist version

3.9.1

### Validations

- [X] Read the [docs](https://perfectionist.dev).
- [X] Check that there isn't [already an issue](https://github.com/azat-io/eslint-plugin-perfectionist/issues) that reports the same bug to avoid creating a duplicate.

**Comment by hugop95** [comment]
@azat-io This is fixed by #324, but that PR will be available in the next major version.

I don't think that this bug appears that often in practice: this requires two imports to be separated by more than the required number of newlines and the additional whitespace, so I don't think that a hotfix in v3.9 is mandatory, but I can make a PR for that if you want as well.

**Comment by azat-io** [comment]
Added in 4369803e38b54f745ca4deefb48d20cdccc0b34b.
Released in [v4.0.0](https://github.com/azat-io/eslint-plugin-perfectionist/releases/tag/v4.0.0).

Thanks!
